### PR TITLE
Add pure python to Getting started

### PR DESCRIPTION
**Pull request by matusvalo**
Partial fix of #4187.

### CODE REVIEW

**Comment by matusvalo** [comment]
Screenshots of sage and jupiter here https://cython.readthedocs.io/en/latest/src/quickstart/build.html were not updated. Not sure whether needed.

**Comment by 0dminnimda** [comment]
> Screenshots of sage and jupiter here https://cython.readthedocs.io/en/latest/src/quickstart/build.html were not updated. Not sure whether
Yes, needed (and i recommend making them on the last steps when text changes will be approved and agreed)

**Comment by scoder** [comment]
> Screenshots of sage and jupiter here https://cython.readthedocs.io/en/latest/src/quickstart/build.html were not updated. Not sure whether needed.

Would be nice, but not required right now. The Jupyter screenshot says "Cython 0.25.2", so it would appear that we managed to live with an old image for a couple of years without anyone complaining.

File: docs/src/quickstart/cythonize.rst [hunk 1]
```diff
@@ -55,27 +69,40 @@ Typing Functions
...
+In our example above, the argument is assumed to be a C double both inside ``f()``
 and in the call to it, yet a Python ``float`` object must be constructed around the
 argument in order to pass it.
 
-Therefore Cython provides a syntax for declaring a C-style function,
-the cdef keyword:
+Therefore Cython language provides a syntax for declaring a C-style function,
+the ``cdef`` keyword. Moreover, Cython library provides ``@cfunc`` decorator to
+declare C-style function without need of using Cython language. Both aproaches are
+equivalent and produces same C code:
```
Comment (scoder) [inline docs/src/quickstart/cythonize.rst:79]:
```suggestion
Therefore, Cython provides a way for declaring a C-style function,
the Cython specific ``cdef`` statement, as well as the ``@cfunc`` decorator to
declare C-style functions in Python syntax.  Both approaches are
equivalent and produce the same C code:
```

File: docs/src/quickstart/cythonize.rst [hunk 2]
```diff
@@ -55,27 +69,40 @@ Typing Functions
...
+
+        .. literalinclude:: ../../examples/quickstart/cythonize/cdef_keyword.pyx
 
 Some form of except-modifier should usually be added, otherwise Cython
 will not be able to propagate exceptions raised in the function (or a
 function it calls). The ``except? -2`` means that an error will be checked
 for if ``-2`` is returned (though the ``?`` indicates that ``-2`` may also
-be used as a valid return value).
+be used as a valid return value). The same can be expressed using only python
+language by ``@exceptval(-2, check=True)`` decorator.
```
Comment (scoder) [inline docs/src/quickstart/cythonize.rst:96]: Please mind the spelling of "Python" (and Cython) as a name with an upper case initial.
```suggestion
be used as a valid return value). The same can be expressed using only Python
syntax with the decorator ``@exceptval(-2, check=True)``.
```

File: docs/src/quickstart/cythonize.rst [hunk 3]
```diff
@@ -84,7 +111,8 @@ typed values directly) and from Python (wrapping values in Python
 objects). In fact, ``cpdef`` does not just provide a Python wrapper, it also
 installs logic to allow the method to be overridden by python methods, even
 when called from within cython. This does add a tiny overhead compared to ``cdef``
-methods.
+methods. Again, Cython provides ``@ccall`` decorator which provides same
```
Comment (0dminnimda) [inline docs/src/quickstart/cythonize.rst:114]:
```suggestion
methods. Also Cython provides ``@ccall`` decorator which provides same
```

Or at least not "again"

File: docs/src/quickstart/cythonize.rst [hunk 4]
```diff
@@ -55,27 +69,40 @@ Typing Functions
...
 for if ``-2`` is returned (though the ``?`` indicates that ``-2`` may also
-be used as a valid return value).
+be used as a valid return value). The same can be expressed using only python
+language by ``@exceptval(-2, check=True)`` decorator.
+
 Alternatively, the slower ``except *`` is always
 safe. An except clause can be left out if the function returns a Python
 object or if it is guaranteed that an exception will not be raised
-within the function call.
+within the function call. Again, Cython provides decorator ``@exceptval(check=True)``
```
Comment (scoder) [inline docs/src/quickstart/cythonize.rst:101]:
```suggestion
within the function call.  Again, Cython provides the decorator ``@exceptval(check=True)``
```

File: docs/src/quickstart/cythonize.rst [hunk 5]
```diff
@@ -55,27 +69,40 @@ Typing Functions
...
+providing the same functionality.
 
-A side-effect of cdef is that the function is no longer available from
-Python-space, as Python wouldn't know how to call it. It is also no
+A side-effect of ``cdef`` (and ``@cfunc`` decorator) is that the function is no longer
+available from Python-space, as Python wouldn't know how to call it. It is also no
```
Comment (scoder) [inline docs/src/quickstart/cythonize.rst:105]:
```suggestion
A side-effect of ``cdef`` (and the ``@cfunc`` decorator) is that the function is no longer
visible from Python-space, as Python wouldn't know how to call it.  It is also no
```

File: docs/src/quickstart/cythonize.rst [hunk 6]
```diff
@@ -84,7 +111,8 @@ typed values directly) and from Python (wrapping values in Python
 objects). In fact, ``cpdef`` does not just provide a Python wrapper, it also
 installs logic to allow the method to be overridden by python methods, even
 when called from within cython. This does add a tiny overhead compared to ``cdef``
-methods.
+methods. Again, Cython provides ``@ccall`` decorator which provides same
+functionality as ``cpdef`` keyword.
```
Comment (scoder) [inline docs/src/quickstart/cythonize.rst:115]:
```suggestion
methods.  Again, Cython provides the ``@ccall`` decorator which provides the same
functionality as ``cpdef`` keyword.
```

**Review by 0dminnimda — CHANGES_REQUESTED** [review]
Also, `cdef_keyword.pyx` will not align, there's need to add empty lines

File: docs/src/quickstart/cythonize.rst (hunk 3 above)
Comment (scoder) [reply docs/src/quickstart/cythonize.rst:114]: What's wrong with "again" here? It was mentioned before.

Comment (0dminnimda) [reply docs/src/quickstart/cythonize.rst]: It's first time mention of ccall in this document, again should refer to something we saw/read

File: docs/src/quickstart/build.rst [hunk 7]
```diff
@@ -3,7 +3,7 @@ Building Cython code
 
 Cython code must, unlike Python, be compiled. This happens in two stages:
 
- - A ``.pyx`` file is compiled by Cython to a ``.c`` file, containing
+ - A ``.pyx`` or ``.py`` file is compiled by Cython to a ``.c`` file, containing
```
[the remark this answers is not in the document]
Comment (matusvalo) [reply docs/src/quickstart/build.rst:6]: There is only one example in jupyter notebook. I considered it as not needed there.

**Comment by matusvalo** [comment]
> Also, cdef_keyword.pyx will not align, there's need to add empty lines
done. Thanks!

File: docs/src/quickstart/cythonize.rst [hunk 8]
```diff
@@ -84,7 +111,8 @@ typed values directly) and from Python (wrapping values in Python
 objects). In fact, ``cpdef`` does not just provide a Python wrapper, it also
 installs logic to allow the method to be overridden by python methods, even
 when called from within cython. This does add a tiny overhead compared to ``cdef``
-methods.
+methods.  Again, Cython provides the ``@ccall`` decorator which provides the same
```
Comment (scoder) [inline docs/src/quickstart/cythonize.rst:114]:
```suggestion
methods.  Again, Cython provides a ``@ccall`` decorator which provides the same
```

**Comment by scoder** [comment]
Thanks

**Pull request merged**